### PR TITLE
feature: add listener mode to service

### PR DESCRIPTION
**Pull request by beav**
This commit adds a new `SERVICE_MODE` env var. If it is set to `REST`,
the service will start as a rest service. This is the default if
`SERVICE_MODE` is not set.

If the `SERVICE_MODE` is set to `LISTENER`, the service will start as a
kafka listener. It can listen for either archival events or delete
events, based on `LISTENER_TYPE`.

This commit moves the code from system-profile-archiver into the
service. The service already has an ORM, and it's cleaner to re-use
this code than to re-implement it as raw SQL in the old archiver.

This commit also removes the ability to POST to create new profiles.
Profiles can only be created via the listener.

Unit test coverage has been moved to 65%. This will be re-raised after
I re-add the profile creation tests.

### CODE REVIEW

File: kafka_listener.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                            "wrote inventory_id %s's profile to historical database"
+                            % host["id"]
+                        )
+                    elif config.listener_mode == "DELETER":
+                        if data.value["type"] == "delete":
+                            inventory_id = data.value["id"]
+                            db_interface.delete_hsps_by_inventory_id(inventory_id)
+                            logger.info(
+                                "deleted profiles for inventory_id %s" % inventory_id
+                            )
```
Comment (ehelms) [inline kafka_listener.py]: It would be nice if the code for archiver and deleter were broken out into their own functions to encapsulate the logic. This big try-if-else block gets hard to parse.

File: historical_system_profiles/config.py [hunk 2]
```diff
@@ -14,3 +14,10 @@
 bootstrap_servers = os.getenv("BOOTSTRAP_SERVERS", "kafka:29092").split(",")
 consume_topic = os.getenv("CONSUME_TOPIC", None)
 group_id = os.getenv("GROUP_ID", None)
+listener_mode = os.getenv("LISTENER_MODE", "REST_SERVICE")
```
Comment (ehelms) [inline historical_system_profiles/config.py]: I would find `service_mode` easier to grok as I don't immediately think that this is related to the listener portion of the code.

Comment (ehelms) [reply historical_system_profiles/config.py]: This might make the code slightly more complex,  but in addition I would find it easier to understand if there was:

service_mode: listener service or api service
listener_type: deleter or archiver

What will the names of the deployments be?

Comment (beav) [reply historical_system_profiles/config.py]: Updated per comment.

Right now the names are "system-profile-archiver", "system-profile-deleter" and "historical-system-profile-backend". I'll probably change the first two to "historical-system-profile-archiver" and "historical-system-profile-deleter".

File: run_app.sh [hunk 3]
```diff
@@ -16,7 +16,16 @@ if [ -z "$PORT" ];
   then PORT=8080
 fi
 
-prometheus_multiproc_dir=$TEMPDIR gunicorn wsgi -w $NUM_WORKERS --threads $THREADS -b 0.0.0.0:$PORT --log-level=$LOG_LEVEL --access-logfile=- --config ./gunicorn.conf.py
+if [ -z "$SERVICE_MODE" ];
+  then SERVICE_MODE=REST
```
Comment (ehelms) [inline run_app.sh]: Sorry for the churn, would you consider `REST_API` as REST alone isn't obvious to me that is the API its spinning up.
